### PR TITLE
link navigation and page traversal

### PR DESCRIPTION
**Pull request by caseyyee**
This hooks up @cvan's hyperlink work with @dmarcos cursor.

navigate to hyperlinks.html to in action.

### CODE REVIEW

**Comment by caseyyee** [comment]
WIP, do not merge yet.    fixing a few issues first.

**Comment by caseyyee** [comment]
OK, this is ready @dmarcos r?

File: examples/cube.html [hunk 1]
```diff
@@ -28,6 +31,29 @@
...
+
+      window.top.postMessage({type: 'ready'}, '*');
+
+      document.getElementById('cube').addEventListener('click', function (e) {
+        var href = e.target.getAttribute('href');
+        if (href) {
+          window.top.postMessage({type: 'navigate', data: {url: href}}, '*');
+        }
+      });
+    }
```
Comment (caseyyee) [inline examples/cube.html]: Will work on moving the above code into either vr-scene, the add-on, or into it's own library.    Once we get deeper into integrating the add-on, we can make a decision then.

File: examples/space.html [hunk 2]
```diff
@@ -43,19 +43,33 @@
     fsButton.addEventListener('click', function() {
       scene.enterVR();
     });
+
     var cubes = document.querySelectorAll('vr-cube');
     var i;
     for (i = 0; i < cubes.length; ++i) {
       cubes[i].addEventListener('click', function(evt) {
```
Comment (caseyyee) [inline examples/space.html]: This click here interferes with the mouse look.    Will use pointerlock to resolve this issue in a later PR.

File: examples/cube2.html [hunk 3]
```diff
@@ -16,13 +16,16 @@
 
 <body>
   <vr-assets>
-    <vr-geometry id="cube" primitive="Box" width="200" height="200" depth="200"></vr-geometry>
+    <vr-geometry id="cube" primitive="Box" width="200" height="200" depth="200" href="space.html"></vr-geometry>
```
Comment (dmarcos) [inline examples/cube2.html]: Why is this href needed?

**Comment by dmarcos** [comment]
Can we leave the cube, space and cube2 examples simple as they are and create new examples to illustrate the links?

**Comment by dmarcos** [comment]
Would it make sense to create a vr-frame component based on VRLoader that it's an iframe on steroids?

**Comment by jcarpenter** [comment]
@caseyyee I agree with dmarcos; let's leave cube, cube2 and space as they were so that the examples directory stays relatively clean, and move our demos into a new demo folder. I can take care of that.

**Comment by caseyyee** [comment]
OK guys, all the VR handling code has been moved into the components along with the Enter VR button.   The example code is now even cleaner and simpler.

**Pull request closed without merging**